src/luarocks/fs/win32.lua

```diff
@@ -336,7 +336,11 @@ function win32.make_temp_dir(name_pattern)
 end
 
 function win32.tmpname()
-   return os.getenv("TMP")..os.tmpname()
+   local tmpname = os.tmpname()
+   if tmpname:match("^\\") then
```

Comment:
```suggestion
   if tmpname:match("\\") then
```
I think the temp name can also be prefixed with a drive letter, eg. `C:\something` in which case this would fail. The old Windows behaviour was anyway that it would only return a filename, and never include a path. So imo checking for a `\` should suffice, just not anchored at the first position.

Reply:
Most of the references I found around mentions the temporary files created in the root directory, so the `\` anchored at the beginning seems the best choice.

Alternatively, another solution is checking if the path returned by `os.tmpname()` starts with `os.getenv('TMP')` and modifying it accordingly.

Reply:
Most of the references I found around mentions the temporary files created in the root directory, so the `\` anchored at the beginning seems the best choice.

Alternatively, another solution is checking if the path returned by `os.tmpname()` starts with `os.getenv('TMP')` and modifying it accordingly.